joschahenningsen/TUM-Live
# Enh/rtmp proxy

### Motivation and Context
See #1386.

### Description
- Updated token panel that allows lecturers to create new _lecturer tokens_ with additional info for different streaming software/services.
- New API endpoint at `/api/token/proxy` that can be used by the RTMP Proxy to exchange the _lecturer token_ with a stream-specific URL for an upcoming stream.
- Added test courses that are automatically created under the "TEST COURSES" semester when one starts streaming without having scheduled active or soon starting stream.

### Steps for Testing
Prerequisites:
- 1 Lecturer
- 1 RTMP-Proxy (https://github.com/arut/nginx-rtmp-module)

1. Log in.
2. Create a soon-starting stream.
3. Go to the token panel and create a new _lecturer token_.
4. Follow the instructions on the screen to start self-streaming from OBS via the RTMP-Proxy, or make a manual POST request to `http://localhost:8081/api/token/proxy/<your-token>` to check if the stream URL is correct.

> [!note]
To change the default settings of the test courses, see: https://github.com/TUM-Dev/gocast/blob/bf61b6d6a8f1b3cbe28f84ad0868fd4f8afb28c2/dao/streams.go#L356

### Screenshots
![image](https://github.com/user-attachments/assets/b77ac408-ed13-4d5a-88b9-d06694d31539)
![image](https://github.com/user-attachments/assets/907252ee-59fc-4597-93f8-79b8859c3478)
![image](https://github.com/user-attachments/assets/35147945-b5af-4159-927b-72439c51bc89)
![image](https://github.com/user-attachments/assets/aadb0682-f90a-4b4e-93be-56b374b8a2d7)
